### PR TITLE
add gen check

### PR DESCRIPTION
**Pull request by zirain**
fixes: https://github.com/envoyproxy/gateway/issues/213

### CODE REVIEW

File: tools/make/lint.mk [hunk 1]
```diff
@@ -64,3 +64,7 @@ lint: lint.shellcheck
 lint.shellcheck:
 	@echo Running Shellcheck linter ...
 	@shellcheck tools/hack/*.sh
+
+.PHONY: gen-check
+gen-check: generate manifests
+	tools/hack/gen-check.sh
```
Comment (arkodg) [inline tools/make/lint.mk:70]: wdyt of skipping the script and adding the check in the Makefile itself ?
```
	@if [ ! -z "`git status --porcelain`" ]; then \
		@echo "ERROR: Some files need to be updated, please run 'make generate' and `make manifests` to include any changed files to your PR"; \
		git diff --exit-code; \
	fi
```

Comment (zirain) [reply tools/make/lint.mk:70]: it's ok for now, let's create a new shell file if there're too much lines in the future.

**Comment by zirain** [comment]
wait for https://github.com/envoyproxy/gateway/pull/354 merged.

**Comment by zirain** [comment]
I will revert the 5bb2c9fec5163f8df10c70a44056a3e2550ed58a once #354 merged.

**Comment by zirain** [comment]
@arkodg this PR is ready for reviews.

**Comment by danehans** [comment]
@zirain please rebase when you have a moment.

File: .github/workflows/build_and_test.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ jobs:
       # lint
       - run: make lint-deps
       - run: make -k lint
+      # gen-check
+      - run: make -k gen-check
```
Comment (LukeShu) [inline .github/workflows/build_and_test.yaml:26]: Can this go in its own job, instead of in the `lint` job?

Comment (arkodg) [reply .github/workflows/build_and_test.yaml:26]: I'd vote for option 3. adding it as a dependency to `lint`. As a dev I dont want to learn more make targets, would be ideal if `make lint` took care of this for me

Comment (zirain) [reply .github/workflows/build_and_test.yaml:26]: @LukeShu @arkodg do your guys make a conclusion？

Comment (arkodg) [reply .github/workflows/build_and_test.yaml:25]: Please ignore my comment and consider Luke's

Comment (zirain) [reply .github/workflows/build_and_test.yaml]: done

**Review by arkodg — APPROVED** [review]
🚀

**Pull request merged**